node/clients/profile/profileV2.ts

```diff
@@ -0,0 +1,303 @@
+
+    return this.get<ProfileV2>(
+      url,
+      {
+        metric: 'profile-system-v2-getProfileInfo',
+        params: {
+          extraFields: customFields,
+        },
+      }
+    ).then((profile: ProfileV2[]) => {
+      if (profile.length > 0) {
+        let profileV2 = profile[0].document
+        profileV2.pii = true
+        profileV2.id = profile[0].id
+        profileV2.isNewsletterOptIn = profileV2.isNewsletterOptIn || false
+
+        return profileV2
+      }
+
+      return {} as Profile
+    })
+  }
+
+  public createProfile = (profile: Profile) =>
+    this.post<ProfileV2>(`${this.baseUrl}?an=${this.account}`, profile)
+      .then((profileV2: ProfileV2) => {
+        let newProfile = profileV2.document as Profile
+        newProfile.id = profileV2.id
+        return newProfile
+      })
+
+  public updatePersonalPreferences = (
+    user: CurrentProfile,
+    personalPreferences: PersonalPreferences
+  ) => {
+    let { userKey, alternativeKey } = this.getUserKeyAndAlternateKey(user)
+    let parsedPersonalPreferences = Object.keys(personalPreferences).reduce((preferences: any, key) => {
+      preferences[key] = personalPreferences[key] === 'True'
+      return preferences
+    }, {})
+
+    return this.patch(
+      `${this.baseUrl}/${userKey}?an=${this.account}&alternativeKey=${alternativeKey}`,
+      parsedPersonalPreferences,
+      {
+        metric: 'profile-system-v2-subscribeNewsletter',
+      }
+    )
+  }
+
+  public updateProfileInfo = (
+    user: CurrentProfile,
+    profile: Profile | { profilePicture: string },
+    customFields?: string
+  ) => {
+    let { userKey, alternativeKey } = this.getUserKeyAndAlternateKey(user)
+
+    if (!(profile as Profile)) {
+      const profileCast = profile as Profile
+      profileCast.gender = profileCast.gender || ""
+      profileCast.document = profileCast.document || ""
+    }
+
+    return this.patch(
+      `${this.baseUrl}/${userKey}?an=${this.account}&alternativeKey=${alternativeKey}`,
+      profile,
+      {
+        metric: 'profile-system-v2-updateProfileInfo',
+        params: {
+          extraFields: customFields,
+        },
+      }
+    )
+  }
+
+  public getUserAddresses = (_: CurrentProfile, currentUserProfile: Profile, piiRequest?: PIIRequest) => {
+    let url = this.getPIIUrl(`${this.baseUrl}/${currentUserProfile.id}/addresses`, undefined, piiRequest)
+
+    return this.get<Address[]>(url, { metric: 'profile-system-v2-getUserAddresses', })
+      .then((addresses: AddressV2[]) => this.translateToV1Address(addresses))
+      .catch<any>(e => {
+        const { response } = e as AxiosError
+        const { status } = response!
+        if (status == 404) {
+          return [] as AddressV2[]
+        }
+
+        return statusToError(e)
+      })
+  }
+
+  private translateToV1Address = (addresses: AddressV2[]) =>
+    addresses.map((address: AddressV2) => {
+      let addressV2 = address.document
```

Comment:
transform to an object, with const or return it directly.

Reply:
Changed it to a `const`. Thanks!